cpp/tests/test_utils.h

```diff
@@ -62,6 +62,23 @@ extern "C" {
 }
 #endif
 
+#define NCCLCHECK(cmd) do {                                   \
+    ncclResult_t nccl_status = cmd;                           \
+    if (nccl_status!= ncclSuccess) {                          \
+      printf("NCCL failure %s:%d '%s'\n",                     \
+          __FILE__,__LINE__,ncclGetErrorString(nccl_status)); \
+    }                                                 \
+  } while(0)
+
+#define MPICHECK(cmd) do {                          \
+  int e = cmd;                                      \
+  if( e != MPI_SUCCESS ) {                          \
+    printf("Failed: MPI error %s:%d '%d'\n",        \
+        __FILE__,__LINE__, e);   \
+    exit(EXIT_FAILURE);                             \
```

Comment:
> Will this cause the entire gtest binary to exit out, possibly before it gets to run any other tests?

Yes

> Maybe this should be a gtest assertion instead, so any test using this macro will just fail with a proper failure message instead of exiting the entire run (assuming that's what will happen).

MPI errors associated with MPI_COMM_WORLD are fatal by default so this macro shouldn't have much impact here. 
Notice that this is just a specific test macro and that the stand-alone test is just checking that the infra can access MPI/NCCL. 
As we add OPG code inside cugraph (ie. as a feature) we should have an OPG error handling. This should be as part of the RAFT communicator abstraction classes. Eventually, we will test OPG cuGraph features using the proper communicator and proper error handling.